source/common/upstream/logical_dns_cluster.cc

```diff
@@ -20,12 +20,20 @@ LogicalDnsCluster::LogicalDnsCluster(const Json::Object& config, Runtime::Loader
           std::chrono::milliseconds(config.getInteger("dns_refresh_rate_ms", 5000))),
       tls_(tls), tls_slot_(tls.allocateSlot()), initialized_(false),
       resolve_timer_(dispatcher.createTimer([this]() -> void { startResolve(); })) {
-
   std::vector<Json::ObjectSharedPtr> hosts_json = config.getObjectArray("hosts");
   if (hosts_json.size() != 1) {
     throw EnvoyException("logical_dns clusters must have a single host");
   }
 
+  // Resolve string to enum.
+  std::string dns_lookup_family = config.getString("dns_lookup_family", "v4_only");
+  if (dns_lookup_family == "v6_only") {
+    dns_lookup_family_ = Network::DnsLookupFamily::v6_only;
+  } else if (dns_lookup_family == "fallback") {
+    dns_lookup_family_ = Network::DnsLookupFamily::fallback;
+  } else {
+    dns_lookup_family_ = Network::DnsLookupFamily::v4_only;
```

Comment:
for being on a defensive side can you assert here that dns_lookup_family is equal to "v4_only".